### ISSUE TITLE
Creating a new class in the assets/scripts folder doesn't set the namespace

### ISSUE THREAD

**Issue by JurjenBiewenga** [JetBrains/resharper-unity#1250]
When creating a new class in the Assets/Scripts folder using the Unity explorer in Rider doesn't fill in the namespace (defaults to DefaultNamespace) correctly even though the root namespace is set in Unity. This does work when done in a sub folder of the scripts folder.

**Comment by van800** [comment]
I expect it is the same problem as https://github.com/JetBrains/resharper-unity/issues/383
When folder is not yet in csproj, Rider can't guess the namespace.

**Comment by JurjenBiewenga** [comment]
Assets/Scripts most definitely is in my csproj as I also have other code there and it wasn't just created. I've talked with Matt about this and he thinks it isn't the same issue.

**Comment by van800** [comment]
Thanks for clarification!

**Comment by citizenmatt** [comment]
This might be because you've got an existing file in that folder, or a parent folder with a namespace of `DefaultNamespace`. 

ReSharper tries to use the same namespace as in your existing files. It looks in the current folder and tries to find the one namespace that is used in all files, ignoring those that don't have a namespace. So if one file has `DefaultNamespace` and the other files don't have any namespace, your new file will get `DefaultNamespace`.

If it can't find a namespace in use in the current folder, it recursively walks up the list of parents, doing the same thing - looking for a namespace that all files use, ignoring files in the global namespace.

If it does find a namespace, it appends the intermediate folder names, unless that folder's "namespace provider" setting is set to `false`, in which case nothing is added. In this case, it's possible for your new file to get `DefaultNamespace` because the parent folder's namespace is determined to be `DefaultNamespace` and your current folder isn't a namespace provider. For Unity, this includes `Assets` and `Assets\Scripts`.

If it still can't find a namespace by looking at files, it will look at the default namespace setting in the `.csproj`, appending folder names, for folders that are namespace providers. If there is no namespace, 
it will use `DefaultNamespace`.

In other words, in testing, this looks like it's working as expected. Check for existing files in the current folder or parent folders that have a namespace of `DefaultNamespace`. That might be overriding the root namespace + folder path value.

**Comment by JurjenBiewenga** [comment]
I just checked and I don't have a namespace in the same folder called `DefaultNamespace`. Can I do anything to help you debug this?
